### PR TITLE
feature/96: Added validation of README.md

### LINKED ISSUE #96: Validate that README contains a link to dependencies.md
PK validates that README contains a link to dependencies.md

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #96

### CODE REVIEW

File: src/main/java/com/exasol/projectkeeper/validators/ReadmeValidator.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    private String getBuildBadge() {
+        return getBadge("Build Status",
+                "https://github.com/exasol/" + this.repoName + "/actions/workflows/ci-build.yml/badge.svg",
+                "https://github.com/exasol/" + this.repoName + "/actions/workflows/ci-build.yml");
+    }
+
+    private String getMavenCentralBadge() {
+        return getBadge("Maven Central", "https://img.shields.io/maven-central/v/com.exasol/" + this.artifactId,
+                "https://search.maven.org/artifact/com.exasol/project-keeper-maven-plugin" + this.artifactId);
```
Comment (morazow) [inline src/main/java/com/exasol/projectkeeper/validators/ReadmeValidator.java:104]: This seems to be typo.
```suggestion
                "https://search.maven.org/artifact/com.exasol/" + this.artifactId);
```

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/projectkeeper/validators/AbstractFileValidator.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                throw new IllegalStateException(ExaError.messageBuilder("E-PK-63")
+                        .message("Failed to create required file {{file path}}.", this.relativeFilePath).toString());
+            }
+        };
+    }
+
+    /**
+     * Create the file with a template content. The {@link AbstractFileValidator} calls this method when it applies the
+     * fix for a missing file.
+     *
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/AbstractFileValidator.java:73]:
```suggestion
     *  Create a new file with a template content. 
     * <p>
     * The {@link AbstractFileValidator} calls this method when it applies the fix for a missing file.
     * </p>
```

File: src/main/java/com/exasol/projectkeeper/ProjectKeeperFixMojo.java [hunk 3]
```diff
@@ -16,6 +21,21 @@
...
+        final List<ValidationFinding> unfixedFindings = new ArrayList<>();
+        getValidators().forEach(validator -> validator.validate().forEach(finding -> {
+            if (finding.hasFix()) {
+                finding.getFix().fixError(log);
+            } else {
+                unfixedFindings.add(finding);
+            }
+        }));
+        for (final ValidationFinding unfixedFinding : unfixedFindings) {
+            log.warn("Could not auto-fix: " + unfixedFinding.getMessage());
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/ProjectKeeperFixMojo.java]: yes. Added

File: src/main/java/com/exasol/projectkeeper/validators/AbstractFileValidator.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        .message("Failed to create required file {{file path}}.", this.relativeFilePath).toString());
+            }
+        };
+    }
+
+    /**
+     *  Create a new file with a template content. 
+     * <p>
+     * The {@link AbstractFileValidator} calls this method when it applies the fix for a missing file.
+     * </p>
```
Comment (morazow) [inline src/main/java/com/exasol/projectkeeper/validators/AbstractFileValidator.java:74]: :+1:

**Review by morazow — APPROVED** [review]
Looks good!

**Pull request merged**